authentikos/authentikos.go

```diff
@@ -218,24 +264,54 @@ func loadClusterConfig() (*rest.Config, error) {
 
 // getOauthTokenCreator returns a function that creates/refreshes an oauth token.
 func getOauthTokenCreator(o options) (tokenCreator, error) {
+	var create tokenCreator
+
 	clientOpts := []option.ClientOption{option.WithScopes(o.scopes...)}
 
 	if len(o.creds) > 0 {
 		clientOpts = append(clientOpts, option.WithCredentialsFile(o.creds))
 	}
 
+	// Reusing the client leverages the token source cache.
 	client, err := transport.Creds(context.Background(), clientOpts...)
-	if err != nil {
-		return nil, err
+
+	clientCreator := func(forceRefresh bool) (*google.Credentials, error) {
+		if forceRefresh {
+			// Recreating the client invalidates the token source cache.
+			client, err = transport.Creds(context.Background(), clientOpts...)
+			printVerbose("force refreshing token\n", o.verbose)
+		}
+
+		return client, err
 	}
 
-	return func() ([]byte, error) {
+	create = func(forceRefresh bool, tries int) ([]byte, error) {
+		if tries <= 0 {
+			return nil, fmt.Errorf("maximum tries: %d exceeded to force refresh token", maxTries)
+		}
+
+		client, err := clientCreator(forceRefresh)
+		if err != nil {
+			return withBackoff(1, maxTries-tries, create).(tokenCreator)(forceRefresh, tries-1)
+		}
+
 		token, err := client.TokenSource.Token()
 		if err != nil {
-			return nil, err
+			return withBackoff(1, maxTries-tries, create).(tokenCreator)(forceRefresh, tries-1)
+		}
+
+		nextRec := time.Now().Add(o.interval * time.Minute)
```

Comment:
This needs to be updated now that the interval has changed types (multiplying by `time.Minute` is now wrong). Please make sure this is covered by unit or integration checks.

Reply:
Good catch. I added a unit test here. Will contrive a integration test in other PR.